job.go

```diff
@@ -207,3 +207,18 @@ func (j *Job) String() string {
 		"{project=%s params=%s group=%s id=%s}",
 		j.Project, j.Params, j.Group, j.ID[:7])
 }
+
+// State determines the job's current state by using it's path in the filesystem.
+func (j *Job) State(path, project, id string) (string, error) {
+	pPath := filepath.Join(path, project, "pending", id)
+	rPath := filepath.Join(path, project, "ready", id)
+	_, err := os.Stat(pPath)
+	if err == nil {
+		return "pending", nil
+	}
+	_, err = os.Stat(rPath)
+	if err == nil {
+		return "ready", nil
+	}
+	return "", err
```

Comment:
I don't like the fact that we may return an empty string and no error. What would such a case indicate?

Reply:
Correction: we do return a non-nil error here, but still, what does an empty string and a "ready path doesn't exist" error means?

Reply:
Thanks for noticing, maybe we could return a custom error then in which we have concatenated any errors with a meaningful message, e.g. "state not found for job %s error %v"

Reply:
A simple `"job %s not found error", j` I think is sufficient.

Reply:
Done